api/v2/views/provider.py

```diff
@@ -5,28 +5,26 @@
 from core.models import Provider, Group
 from core.query import only_current_provider, only_current
 
-from api.permissions import CloudAdminRequired
 from api.v2.serializers.details import ProviderSerializer
+from api.v2.serializers.post import ProviderSerializer as POST_ProviderSerializer
 from api.v2.serializers.summaries import SizeSummarySerializer
-from api.v2.views.base import AuthReadOnlyViewSet
+from api.v2.views.base import AuthViewSet
 from api.v2.views.mixins import MultipleFieldLookup
 
 
-class ProviderViewSet(MultipleFieldLookup, AuthReadOnlyViewSet):
+class ProviderViewSet(MultipleFieldLookup, AuthViewSet):
     """
     API endpoint that allows providers to be viewed or edited.
     """
     lookup_fields = ("id", "uuid")
     queryset = Provider.objects.all()
     serializer_class = ProviderSerializer
-    http_method_names = ['get', 'head', 'options', 'trace']
+    http_method_names = ['get', 'post', 'head', 'options', 'trace']
 
```

Comment:
Potentially add permission class for staff/superuser/cloud-admin here and in `v2/views/accounts.py`